crates/edr_utils/src/types.rs

```diff
@@ -0,0 +1,4 @@
+// Trait for retrieving a sub-type from a higher-kind type.
+pub trait HigherKinded<ParamT> {
+    type Type;
+}
```

Comment:
Hey thanks for the PR this, it's a big achievement! 

I had multiple passes now and I think I'm starting to get it, but tbh this trait still confuses me, so bear with me if I'm off the mark here:

My understanding is that the trait is used to map execution log types to execution receipt types (execution log types are associated types in execution receipt types). Is this correct? If yes, I'd maybe change the name to `InverseAssociatedType` or something like that?

A higher kinded type for me would be something like a functor-like generic map trait that we implement for execution receipt types instead of using the custom `MapReceiptLogs` trait.

Reply:
I had to look this up myself as I wasn't 100% sure, but my understanding is "Higher kinded types are types which take other types and construct a new type".

In this case, we are indeed constructing the "inverse associated type", but the `trait HigherKinded` can be used to generate any type based on a type parameter.

The `ExecutionReceiptHigherKindedForChainSpec` is an implementation that constructs a chain-specific execution receipt based on the log type.

Reply:
We discussed offline and came up with the [following changes](https://github.com/NomicFoundation/edr/pull/741/commits/dc4bc44c380dbf568f098299b88a45094556637b) to address the concern.